romamyronenko/python_messenger_api
# 19 add funcs and endpoints to work with db

Summary of Changes:
Add test for sending messages: Implemented a test to ensure the send_message endpoint correctly processes and stores messages in the database.
Add test for retrieving messages: Implemented a test to verify that the get_messages endpoint returns the correct messages for a specific chat.
Details:
Test for send_message endpoint:

Sends a message to a specific chat using the POST /chat/{chat_id}/message endpoint.
Validates that the response includes the correct message details (conversation_id, message_text, user_id).
Ensures that the message is saved in the database and matches the sent data.
Test for get_messages endpoint:

Adds a test message to the database before calling the GET /chat/{chat_id}/message endpoint.
Validates that the response contains the correct message details in a list format and matches the sent data.
Ensures that the correct status code (200) is returned for a valid chat.
How to Test:
Run pytest to execute the test cases.
Verify that the tests for sending and retrieving messages pass successfully.
Changes in the Codebase:
Added the test_send_message function in tests/test_messages.py for testing the message sending functionality.
Added the test_get_message function in tests/test_messages.py for testing the message retrieval functionality.
Related Issues:
No related issues.
Impact:
Adds unit tests for the message endpoints, ensuring the backend works correctly for sending and retrieving messages from a chat.
